src/adc121c021/adc121c021.cxx

```diff
@@ -57,6 +58,80 @@ ADC121C021::ADC121C021(int bus, uint8_t address, float vref)
   m_vref = vref;
 }
 
+ADC121C021::ADC121C021(std::string initStr) : mraaIo(initStr)
+{
+  mraa_io_descriptor* descs = mraaIo.getMraaDescriptors();
+
+  std::vector<std::string> upmTokens;
+
+  if (!mraaIo.getLeftoverStr().empty()) {
+      upmTokens = UpmStringParser::parse(mraaIo.getLeftoverStr());
+  }
+
+  if(!descs->i2cs)
+  {
+    throw std::invalid_argument(std::string(__FUNCTION__) +
+                                  ": mraa_i2c_init() failed");
+  }
+  else
+  {
+    if( !(m_i2c = descs->i2cs[0]) )
+    {
+      throw std::invalid_argument(std::string(__FUNCTION__) +
+                                  ": mraa_i2c_init() failed");
+
+    }
+  }
+
+  std::string::size_type sz;
+
+  for (std::string tok : upmTokens) {
+    if(tok.substr(0, 5) == "vref:") {
+        m_vref = std::stof(tok.substr(5));
+    }
+    if(tok.substr(0,10) == "writeByte:") {
+      uint8_t reg = std::stoi(tok.substr(10),&sz,0);
+      tok = tok.substr(10);
+      uint8_t byte = std::stoi(tok.substr(sz+1),nullptr,0);
+      writeByte(reg, byte);
+    }
+    if(tok.substr(0,10) == "writeWord:") {
+      uint8_t reg = std::stoi(tok.substr(10),&sz,0);
+      tok = tok.substr(10);
+      uint16_t word = std::stoi(tok.substr(sz+1),nullptr,0);
+      writeWord(reg, word);
+    }
+    if(tok.substr(0,16) == "enableAlertFlag:") {
+      bool enable = std::stoi(tok.substr(16),nullptr,0);
+      enableAlertFlag(enable);
+    }
+    if(tok.substr(0,15) == "enableAlertPin:") {
+      bool enable = std::stoi(tok.substr(15),nullptr,0);
+      enableAlertPin(enable);
+    }
+    if(tok.substr(0,16) == "enableAlertHold:") {
+      bool enable = std::stoi(tok.substr(16),nullptr,0);
+      enableAlertHold(enable);
+    }
+    if(tok.substr(0,27) == "enableAlertPinPolarityHigh:") {
+      bool enable = std::stoi(tok.substr(27),nullptr,0);
+      enableAlertPinPolarityHigh(enable);
+    }
+    if(tok.substr(0,17) == "setAlertLowLimit:") {
```

Comment:
Some conversions could modify the expected results here. Stoi returns an integer.